Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs

```diff
@@ -576,6 +578,52 @@ public IEnumerator TestInteractableDisableOnClick()
             GameObject.Destroy(interactable.gameObject);
         }
 
+        [UnityTest]
+        /// <summary>
+        /// Tests that the toggle button states consistently return to original state
+        /// after subsequent clicks (front plate does not move back after every click).
+        /// </summary>
+        public IEnumerator TestPressableToggleHoloLens2()
+        {
+            var rightHand = new TestHand(Handedness.Right);
+            Vector3 p2 = new Vector3(0.015f, 0f, 0.3f);
+
+            Interactable interactable;
+            Transform frontPlateTransform;
+
+            InstantiatePressableButtonHoloLens2Toggle(
+                new Vector3(0.0f, 0.1f, 0.4f),
+                DefaultRotationToggle,
+                out interactable,
+                out frontPlateTransform);
+
+            Assert.True(interactable.IsEnabled);
+
+            bool wasClicked = false;
+            interactable.OnClick.AddListener(() => { wasClicked = true; });
+
+            // Get start position of the front plate before button is pressed
+            Vector3 frontPlateStartPosition = frontPlateTransform.localPosition;
+
+            yield return rightHand.Show(p2);
+
+            int numClicks = 3;
+            for (int i = 0; i < numClicks; i++)
+            {
+                wasClicked = false;
+                yield return rightHand.Click();
+                // Wait for button animation to complete
+                yield return new WaitForSeconds(0.33f);
+
+                Assert.True(wasClicked, "Toggle button was not clicked");
```

Comment:
>Assert.True(wasClicked, "Toggle button was not clicked"); [](start = 16, length = 57)

Can you also add validation for CurrentDimension ? (0 on 1st click, 1 on second click and 0 again on 3rd click )

Reply:
That's covered in TestDimensions, not sure it's needed here?